Sources/JWTKeychain/LoginRequest.swift

```diff
@@ -0,0 +1,37 @@
+import Vapor
+
+public protocol LoginRequest: AuthenticationRequest, ValidatableRequest {
+    typealias Model = User
+
+    static var hashedPasswordKey: KeyPath<Model, String> { get }
+
+    var password: String { get }
+
+    func logIn(on request: Request) -> EventLoopFuture<Model>
+}
+
+public extension LoginRequest {
+    static func logIn(
+        on request: Request,
+        errorOnWrongPassword: Error,
+        currentDate: Date = Date()
+    ) -> EventLoopFuture<AuthenticationResponse<Model>> {
+        validated(on: request).flatMap { loginRequest in
+            loginRequest
+                .logIn(on: request)
+                .flatMap { user in
+                    request.password.async
+                        .verify(loginRequest.password, created: user[keyPath: hashedPasswordKey])
+                        .flatMapThrowing { passwordsMatch in
+                            guard passwordsMatch else { throw errorOnWrongPassword }
```

Comment:
I didn't commit this to the Reset repo yet but I think this should be: 
```suggestion
        errorOnWrongPassword: @escaping @autoclosure () -> Error,
        currentDate: Date = Date()
    ) -> EventLoopFuture<AuthenticationResponse<Model>> {
        validated(on: request).flatMap { loginRequest in
            loginRequest
                .logIn(on: request)
                .flatMap { user in
                    request.password.async
                        .verify(loginRequest.password, created: user[keyPath: hashedPasswordKey])
                        .flatMapThrowing { passwordsMatch in
                            guard passwordsMatch else { throw errorOnWrongPassword() }
```
To prevent the error from being initialized unnecessarily (although I'm not sure that will do much harm)